config/config.go

```diff
@@ -1,91 +1,79 @@
-		// Service how must be service called and displayed in Datadog system
-		Service string `mapstructure:"dd_service"`
-		// ServiceVersion depends on system, can be Git Tag or API version
-		ServiceVersion string `mapstructure:"dd_version"`
-		// DSD Socket path for DD StatsD, important to have unix prefix for that value, example: unix:///var/run/dd/dsd.socket
-		DSD string `mapstructure:"dd_dogstatsd_url"`
-		// APM Socket path for apm and profiler, unix prefix not needed, example: /var/run/dd/apm.socket
-		APM string `mapstructure:"dd_trace_agent_url"`
-	}
+    // DatadogParameters for connection and configuring background process to send information to Datadog Agent
+    DatadogParameters interface {
+        // GetEnv where application is executed, dev, production, staging etc
+        GetEnv() string
+        // GetService how must be service called and displayed in Datadog system
+        GetService() string
+        // GetServiceVersion depends on system, can be Git Tag or API version
+        GetServiceVersion() string
+        // GetDsdEndpoint Socket path or URL for DD StatsD
+        GetDsdEndpoint() string
+        // GetApmEndpoint Socket path or URL for APM and profiler
+        GetApmEndpoint() string
+        // IsDataDogConfigValid method to verify if configuration values are correct
+        IsDataDogConfigValid() bool
+    }
+    // DatadogConfig that required to connect to Datadog Agent
+    DatadogConfig struct {
+        // Env where application is executed, dev, production, staging etc
+        Env string `mapstructure:"dd_env"`
+        // Service how must be service called and displayed in Datadog system
+        Service string `mapstructure:"dd_service"`
+        // ServiceVersion depends on system, can be Git Tag or API version
+        ServiceVersion string `mapstructure:"dd_version"`
+        // DSD Socket path for DD StatsD, important to have unix prefix for that value, example: unix:///var/run/dd/dsd.socket
+        DSD string `mapstructure:"dd_dogstatsd_url"`
+        // APM Socket path for apm and profiler, unix prefix not needed, example: /var/run/dd/apm.socket
+        APM string `mapstructure:"dd_trace_agent_url"`
+    }
 )
 
 // IsDataDogConfigValid method to verify if configuration values are correct
 func (d DatadogConfig) IsDataDogConfigValid() bool {
-	if d.Env == "" {
-		return false
-	}
-	if d.Service == "" {
-		return false
-	}
-	if d.ServiceVersion == "" {
-		return false
-	}
-
-	// Check socket paths
-	if d.DSD != "" && !strings.Contains(d.DSD, "unix") {
-		return false
-	}
-	if d.APM != "" && strings.Contains(d.APM, "unix") {
-		return false
-	}
+    if d.Env == "" {
+        return false
+    }
+    if d.Service == "" {
+        return false
+    }
+    if d.ServiceVersion == "" {
+        return false
+    }
 
-	// DSD or APM must be configured`
-	if d.DSD == "" && d.APM == "" {
-		return false
-	}
+    // DSD or APM must be configured`
+    if d.DSD == "" || d.APM == "" {
```

Comment:
This will break for those that might only use APM.

Reply:
But probably okay since it's just one additional env var to define, and it's only not working in Azure at the moment anyway, so fine then.